src/main/java/frc/robot/subsystems/driveTrain/DriveTrainConstants.java

```diff
@@ -0,0 +1,152 @@
+
+    public static final double ANGLE_OFFSET_FL = 92.461;
+    public static final double ANGLE_OFFSET_FR = 9.141;
+    public static final double ANGLE_OFFSET_BL = 93.428;
+    public static final double ANGLE_OFFSET_BR = 89.033;
+
+    public static final int PIGEON2_ID = 0;
+
+    // --------------------------------------------------------
+
+    // kinematics:
+    public static final double WIDTH_METERS = 0.4857;
+    public static final double HEIGHT_METERS = 0.4857;
+    public static final Translation2d TL_TRANSLATION = new Translation2d(WIDTH_METERS / 2, HEIGHT_METERS / 2);
+    public static final Translation2d TR_TRANSLATION = new Translation2d(WIDTH_METERS / 2, -HEIGHT_METERS / 2);
+    public static final Translation2d BL_TRANSLATION = new Translation2d(-WIDTH_METERS / 2, HEIGHT_METERS / 2);
+    public static final Translation2d BR_TRANSLATION = new Translation2d(-WIDTH_METERS / 2, -HEIGHT_METERS / 2);
+
+    public static final SwerveDriveKinematics KINEMATICS = new SwerveDriveKinematics(
+            TL_TRANSLATION,
+            TR_TRANSLATION,
+            BL_TRANSLATION,
+            BR_TRANSLATION
+    );
+
+    // --------------------------------------------------------
+
+    // controllers and drive settings
+
+    public static final double MAX_TURNING_RAD_PS = 1.5 * Math.PI;
+    public static final double MAX_VELOCITY_MPS = 4;
+
+    public static final double MIN_TURNING_RAD_PS = 0.4;
+    public static final double MIN_VELOCITY_MPS = MAX_VELOCITY_MPS * 0.01;
+
+    public static final PIDFTerms DRIVE_MOTOR_PID = new PIDFTerms(0.1, 0, 0, 1023 / 21375);
+    public static final PIDFTerms TURNING_MOTOR_PID = new PIDFTerms(0.3, 0, 0, 0);
+    public static final int TURNING_MOTOR_CRUISE_VEL = 17000;
+    public static final int TURNING_MOTOR_ACC = 40000;
+    public static final int TURNING_MOTOR_ACC_SMOOTHING = 0;
+
+    public static final double DRIVE_SENSITIVITY = 0.5;
+    public static final double ROTATION_SENSITIVITY = 0.5;
+
+    public static final PIDFTerms KEEP_ANGLE_PID = new PIDFTerms(20, 0.07, 0, 0);
+
+    public static final double KEEP_ANGLE_ROTATION_DELAY = 0.5;
+    public static final double KEEP_ANGLE_DRIVE_DELAY = 0.25;
+
+    public static final int MEDIAN_SAMPLES = 3;
+    public static final int MEDIAN_SAMPLES_ROTATION = 5;
+    public static final double SLEW_RATE_LIMITER_DRIVE = 2;
+    public static final int SLEW_RATE_LIMITER_ROTATION = 2;
+
+    public static final PIDFTerms AUTO_PILOT_PID_X = new PIDFTerms(1.6, 0, 0, 0);
+    public static final PIDFTerms AUTO_PILOT_PID_Y = new PIDFTerms(1.6, 0, 0, 0);
+    public static final PIDFTerms AUTO_PILOT_PID_ROT = new PIDFTerms(2, 0, 0, 0);
+    public static double AUTO_PILOT_DRIVE_SENSITIVITY = 0.5;
+    public static double AUTO_PILOT_TURNING_SENSITIVITY = 0.5;
+    public static double AUTO_PILOT_DRIVE_TOLERANCE = 0.03;
+    public static double AUTO_PILOT_ROTATION_TOLERANCE = Math.toRadians(5);
```

Comment:
pleas order this in this order:

1.  single module constrants
2. full chassis constants (speed, radPS) + sensetivitys
3. keep angle
4. auto